packages/ketcher-core/src/application/render/restruct/rebond.ts

```diff
@@ -57,18 +57,167 @@ class ReBond extends ReObject {
     return true
   }
 
+  static bondRecalc(bond: ReBond, restruct: ReStruct, options: any): void {
+    const render = restruct.render
+    const atom1 = restruct.atoms.get(bond.b.begin)
+    const atom2 = restruct.atoms.get(bond.b.end)
+
+    if (
+      !atom1 ||
+      !atom2 ||
+      bond.b.hb1 === undefined ||
+      bond.b.hb2 === undefined
+    ) {
+      return
+    }
+
+    const p1 = Scale.obj2scaled(atom1.a.pp, render.options)
+    const p2 = Scale.obj2scaled(atom2.a.pp, render.options)
+    const hb1 = restruct.molecule.halfBonds.get(bond.b.hb1)
+    const hb2 = restruct.molecule.halfBonds.get(bond.b.hb2)
+
+    if (!hb1?.dir || !hb2?.dir) return
+
+    hb1.p = shiftBondEnd(atom1, p1, hb1.dir, 2 * options.lineWidth)
+    hb2.p = shiftBondEnd(atom2, p2, hb2.dir, 2 * options.lineWidth)
+    bond.b.center = Vec2.lc2(atom1.a.pp, 0.5, atom2.a.pp, 0.5)
+    bond.b.len = Vec2.dist(p1, p2)
+    bond.b.sb = options.lineWidth * 5
+    /* eslint-disable no-mixed-operators */
+    bond.b.sa = Math.max(bond.b.sb, bond.b.len / 2 - options.lineWidth * 2)
+    /* eslint-enable no-mixed-operators */
+    bond.b.angle = (Math.atan2(hb1.dir.y, hb1.dir.x) * 180) / Math.PI
+  }
+
   drawHover(render: Render) {
     const ret = this.makeHoverPlate(render)
     render.ctab.addReObjectPath(LayerMap.hovering, this.visel, ret)
     return ret
   }
 
+  getSelectionPoints(render: Render) {
+    const bond: Bond = this.b
```

Comment:
We render both regular and stereo bonds using the same SVG path, just with slightly different parameters.
Both bond types use two lines and two bezier curves to draw their paths:
<img width="410" alt="Screenshot 2023-05-19 at 15 14 48" src="https://github.com/epam/ketcher/assets/101649391/3536a380-147b-4582-97eb-7521c9446b8c">